frontend/src/ShardsDashboard.jsx

```diff
@@ -1,15 +1,64 @@
 import React from "react";
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Typography from '@material-ui/core/Typography';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import { loadShardStatus } from './data/shards/actions';
+
+const mapStateToProps = ({ shards }) => ({
+	...shards,
+});
+
+const mapDispatchToProps = {
+	loadShardStatus,
+}
 
 class ShardsDashboard extends React.Component {
+	static propTypes = {
+		loadingState: PropTypes.oneOf(['NOT_LOADED', 'LOADING', 'LOADED', 'ERROR']).isRequired,
+		data: PropTypes.object,
+		error: PropTypes.object,
+		loadShardStatus: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		const { loadingState } = this.props;
+		if (loadingState === 'NOT_LOADED' || loadingState === 'ERROR') {
+			this.props.loadShardStatus();
+		}
+	}
+
+	getContainerStyles() {
+		return {
+			display: 'flex',
+			margin: '0 auto',
+			width: '100%',
+			alignItems: 'flex-start',
+			justifyContent: 'center',
+		};
+	}
+
     render() {
-        return (
-            <div>
-                Hello from shards!
-            </div>
-        )
-    }
-}
+		const { loadingState } = this.props;
+
+		if (loadingState === 'NOT_LOADED' || loadingState === 'LOADING') {
```

Comment:
ooooh we can use suspense here (not needed in this PR though)

Reply:
Yeah totally could! Probably for a separate PR though as you say.